### PR TITLE
ROC-5244 Declare dates you can't attend a hearing

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-5244

### Change description
New page on the hearing task to allow users to declare dates they cannot attend a hearing

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [x] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/public/stylesheets/components/_date-picker.scss [hunk 1]
```diff
@@ -0,0 +1,161 @@
+// sass-lint:disable nesting-depth
```
Comment (andrewwa-kainos) [inline src/main/public/stylesheets/components/_date-picker.scss]: @robertparkinson is looking at refactoring this so we don't need the exclusion

File: src/main/features/directions-questionnaire/forms/models/availability.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+export class Availability {
+
+  @IsDefined({ message: GlobalValidationErrors.YES_NO_REQUIRED })
+  hasUnavailableDates?: boolean
+
+  @ValidateIf(o => o.hasUnavailableDates)
+  @ValidateNested()
+  @ArrayNotEmpty({ message: ValidationErrors.AT_LEAST_ONE_DATE })
+  @IsValidLocalDate({ message: ValidationErrors.DATE_NOT_VALID, each: true })
+  unavailableDates?: LocalDate[]
```
Comment (andrewwa-kainos) [inline src/main/features/directions-questionnaire/forms/models/availability.ts]: Requirement from Marc to validate the array _is_ empty when the user says they don't want to declare any.

File: src/main/public/js/date-picker.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    datePicker.selector().datepicker("setDates", d);
+  },
+
+  selector: () => $("#date-picker"),
+
+  setUpDOWHeading: () => {
+    const days = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
+    const dow = $(".dow");
+    $.each(dow, (index) => $(this).text(days[index]));
+  },
```
Comment (andrewwa-kainos) [inline src/main/public/js/date-picker.js]: This seems to have stopped working

Comment (ellie-harrison) [reply src/main/public/js/date-picker.js]:
```
 setUpDOWHeading: () => {
    const days = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
    $(".dow").each((key, value) => {
      $(value).text(days[key])
    });
  }
```
 should fix this

**Pull request merged**